great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4637,7 +4637,11 @@ def _instantiate_datasource_from_config(
                 raw_config=raw_config, substituted_config=substituted_config
             )
         except Exception as e:
-            name = substituted_config.name or ""
+            try:
+                name = substituted_config.name or ""
+            except AttributeError:
+                # dict object has no attribute 'name'
+                name = "(unknown name)"
```

Comment:
You might want to get feedback here from folks working on datasource persistence e.g. @Kilo59

Reply:
At this point in the code, the name is only used to be a part of the Exception message that is going to be raised.

Reply:
Good point, thanks for clarifying!